applications/tari_console_wallet/src/ui/state/tasks.rs

```diff
@@ -215,3 +225,101 @@ pub async fn send_one_sided_to_stealth_address_transaction(
         },
     }
 }
+
+pub async fn send_burn_transaction_task(
+    burn_proof_filepath: Option<PathBuf>,
+    claim_public_key: Option<PublicKey>,
+    amount: MicroTari,
+    selection_criteria: UtxoSelectionCriteria,
+    message: String,
+    fee_per_gram: MicroTari,
+    mut transaction_service_handle: TransactionServiceHandle,
+    db: WalletDatabase<WalletSqliteDatabase>,
+    result_tx: watch::Sender<UiTransactionBurnStatus>,
+) {
+    result_tx.send(UiTransactionBurnStatus::Initiated).unwrap();
+    let mut event_stream = transaction_service_handle.get_event_stream();
+
+    match transaction_service_handle
+        .burn_tari(amount, selection_criteria, fee_per_gram, message, claim_public_key)
+        .await
```

Comment:
This adds an unnecessary nest, 
```suggestion
 let (burn_tx_id, original_proof)= ransaction_service_handle
        .burn_tari(amount, selection_criteria, fee_per_gram, message, claim_public_key)
        .await.map_err(|e|{result_tx
                .send(UiTransactionBurnStatus::Error(UiError::from(e).to_string()))
                .unwrap();})?;
```